tests/ActivityLoggerBatchTest.php

```diff
@@ -0,0 +1,173 @@
+<?php
+
+namespace Spatie\Activitylog\Test;
+
+use Spatie\Activitylog\ActivityLoggerBatch;
+
+class ActivityLoggerBatchTest extends TestCase
+{
+
+    /** @test */
+    public function it_generates_uuid_after_start_and_end_batch_properely()
+    {
+        app(ActivityLoggerBatch::class)->startBatch();
+        $uuid = app(ActivityLoggerBatch::class)->getUuid();
+        app(ActivityLoggerBatch::class)->endBatch();
+
+
+        $this->assertNotEmpty($uuid);
+    }
+
+    /** @test */
+    public function it_returns_null_uuid_after_end_batch_properely()
+    {
+        app(ActivityLoggerBatch::class)->startBatch();
+        $uuid = app(ActivityLoggerBatch::class)->getUuid();
+        app(ActivityLoggerBatch::class)->endBatch();
+
+
+        $this->assertNotNull($uuid);
+        $this->assertNull(app(ActivityLoggerBatch::class)->getUuid());
+    }
+
+
+    /** @test */
+    public function it_generates_a_new_uuid_after_starting_new_batch_properly()
+    {
+        app(ActivityLoggerBatch::class)->startBatch();
+        $firstBatchUuid = app(ActivityLoggerBatch::class)->getUuid();
+        app(ActivityLoggerBatch::class)->endBatch();
+
+        app(ActivityLoggerBatch::class)->startBatch();
+        $secondBatchUuid = app(ActivityLoggerBatch::class)->getUuid();
+        app(ActivityLoggerBatch::class)->endBatch();
+
+
+        $this->assertNotNull($firstBatchUuid);
+        $this->assertNotNull($secondBatchUuid);
+
+        $this->assertNotEquals($firstBatchUuid, $secondBatchUuid);
+    }
+
+
+    /** @test */
+    public function it_will_not_generate_new_uuid_if_start_already_started_batch()
+    {
+        app(ActivityLoggerBatch::class)->startBatch();
+
+        $firstUuid = app(ActivityLoggerBatch::class)->getUuid();
+
+        app(ActivityLoggerBatch::class)->startBatch();
+
+        $secondUuid = app(ActivityLoggerBatch::class)->getUuid();
+
+        app(ActivityLoggerBatch::class)->endBatch();
+
+
+        $this->assertEquals($firstUuid, $secondUuid);
+    }
+
+
+    /** @test */
+    public function it_will_not_generate_uuid_if_end_batch_before_starting()
+    {
+        app(ActivityLoggerBatch::class)->endBatch();
```

Comment:
This end and also the start call aren't important for this test

Reply:
The test says end before starts, and that's exactly what It does.